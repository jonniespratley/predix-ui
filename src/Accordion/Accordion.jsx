import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styled, { css } from '../styled';
import IronCollapse from '../iron/IronCollapse';
import Flex from '../Flex';
import Icon from '../Icon/Icon';

const styles = {
  width: '16px',
  height: '16px'
};

const svgStyles = {
  pointerEvents: 'none',
  display: 'block',
  height: '16px',
  width: '16px',
  fill: 'none',
  stroke: 'currentColor'
};
const CloseIcon = () => (<i className="px-icon px-icon-utl px-utl-chevron" style={styles}><svg viewBox="0 0 16 16" preserveAspectRatio="xMidYMid meet" style={svgStyles}><g id="px-utl-chevron-right"><path strokeLinejoin="round" d="M6.2 13.2l5.4-5.5-5.5-5.5" /></g></svg></i>);
const OpenIcon = () => (<i className="px-icon px-icon-utl px-utl-chevron-down" style={styles}><svg viewBox="0 0 16 16" preserveAspectRatio="xMidYMid meet" style={svgStyles}><g id="px-utl-chevron"><path d="M2.4 6.2l5.5 5.5 5.5-5.5" /></g></svg></i>);

const AccordionHeader = styled.div`
  padding: .5rem;
  margin-bottom   : 0.3333rem;
  cursor          : pointer;
  user-select     : none;

  color: var(--px-headings-heading-subsection-color, black);
  background-color: var(--px-headings-heading-subsection-background, whitesmoke);

  background-color: var(--px-accordion-header-background-color, whitesmoke);
  color           : var(--px-accordion-header-color, black);

  

  display         : flex;
  justify-content : space-between;
  > * {
    user-select   : none;
  }

  ${props => props.disabled && css`
    cursor:not-allowed;
    pointerEvents: none;
  `}
`;
AccordionHeader.displayName = 'AccordionHeader';
AccordionHeader.defaultProps = {
  className: 'px-accordion__header'
};

const AccordionHeaderText = styled.span`
  text-transform:uppercase;
  margin-left: .2rem;
`;
AccordionHeaderText.displayName = 'AccordionHeaderText';

const AccordionContainer = styled.div`
  display: flex;
  flex-direction: column;
  font-family: "GE Inspira Sans";
  -webkit-transition-duration: 300ms;
  -moz-transition-duration: 300ms;
  transition-duration: 300ms;
`;
AccordionContainer.displayName = 'AccordionContainer';
AccordionContainer.defaultProps = {
  className: 'px-accordion__container'
};

const AccordionSection = styled.div`
  display: flex;
  flex-direction: column;
`;
AccordionSection.displayName = 'AccordionSection';
AccordionSection.defaultProps = {
  className: 'px-accordion__section'
};

const AccordionBody = styled.div`
  margin  : 0;
  overflow: auto;
  padding: .5rem;
`;
AccordionBody.displayName = 'AccordionBody';
AccordionBody.defaultProps = {
  className: 'px-accordion__body'
};

const AccordionIcon = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width : 1rem;
  height: 1rem;
`;
AccordionIcon.displayName = 'AccordionIcon';

const AccordionStatus = styled.div`
  margin-right  : 1rem;
  text-transform: none;
`;
AccordionStatus.displayName = 'AccordionStatus';

const AccordionAction = styled.div`
  cursor: pointer;
  color: var(--px-actionable-alt-text-color, blue);
  background: transparent;
  border: transparent;
  outline: none;
  --iron-icon-width: 1rem;
  --iron-icon-height: 1rem;
`;
AccordionAction.displayName = 'AccordionAction';

class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      closed: props.closed
    };
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    const { disabled } = this.props;
    if (!disabled) {
      this.setState(prevState => ({
        closed: !prevState.closed
      }));
    }
  }

  componentDidUpdate() {
    const { onCollapsed, onExpanded } = this.props;
    const { closed } = this.state;
    if (!closed) {
      onExpanded ? onExpanded(this.state) : null;
    } else {
      onCollapsed ? onCollapsed(this.state) : null;
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ closed: nextProps.closed });
  }

  render() {
    const { closed } = this.state;
    const {
      headerText,
      headerValue,
      status,
      disabled,
      actions,
      showAction,
      onActionClick,
      icons,
      children
    } = this.props;
    const { action } = icons;
    const baseClasses = classNames(
      'px-accordion',
      { 'px-accordion--disabled': disabled },
      { 'px-accordion--open': !closed }
    );
    const headerClasses = classNames('px-accordion__header');

    return (
      <AccordionContainer className={baseClasses}>
        <AccordionHeader disabled={disabled} className={headerClasses}>
          <Flex item onClick={this.onClick}>
            <span>
              {!closed && <OpenIcon />}
              {closed && <CloseIcon />}
            </span>
            <AccordionHeaderText>{headerText || headerValue}</AccordionHeaderText>
          </Flex>
          <Flex middle>
            {status && <AccordionStatus>{status}</AccordionStatus>}
            {showAction
            && (
            <AccordionAction onClick={onActionClick}>
              <Icon icon={action} size={16} />
            </AccordionAction>
            )}
            {actions
            && (
            <AccordionAction>
              {actions()}
            </AccordionAction>
            )}
          </Flex>
        </AccordionHeader>
        <IronCollapse opened={!closed}>
          <AccordionBody>
            {children}
          </AccordionBody>
        </IronCollapse>
      </AccordionContainer>
    );
  }
}

Accordion.displayName = 'Accordion';

Accordion.propTypes = {
  headerText: PropTypes.string,
  headerValue: PropTypes.string,
  status: PropTypes.string,
  closed: PropTypes.bool,
  disabled: PropTypes.bool,
  actions: PropTypes.node,
  showAction: PropTypes.bool,
  onExpanded: PropTypes.func,
  onCollapsed: PropTypes.func,
  onActionClick: PropTypes.func,
  icons: PropTypes.shape({
    more: PropTypes.string,
    left: PropTypes.string,
    action: PropTypes.string
  }),
  children: PropTypes.node
};

Accordion.defaultProps = {
  headerText: null,
  headerValue: null,
  status: null,
  closed: null,
  disabled: null,
  actions: null,
  showAction: null,
  onCollapsed: null,
  onExpanded: null,
  onActionClick: null,
  icons: {
    more: 'px-utl:chevron-right',
    left: 'px-utl:chevron',
    action: 'px-utl:edit'
  },
  children: null
};

export default Accordion;
