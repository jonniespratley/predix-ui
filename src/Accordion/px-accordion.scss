// Base
@import "px-flexbox-design/_base.flexbox.scss"; //@import "px-headings-design/_base.headings.scss"; // Objects
@import "px-actionable-design/_objects.actionable.scss"; // Generic
//@import "px-normalize-design/_generic.normalize.scss";
//@import "px-box-sizing-design/_generic.box-sizing.scss";
//@import "px-helpers-design/_generic.helpers.scss";
// Trumps
$inuit-enable-paddings--tiny: true;
@import "px-spacing-design/_trumps.spacing.scss";
/**
 * px-accordion styles
 */
.px-accordion {
  @import "px-flexbox-design/_base.flexbox.scss"; //@import "px-headings-design/_base.headings.scss"; // Objects
  @import "px-actionable-design/_objects.actionable.scss";

  .accordion {
    &__header {
      margin-bottom   : 0.3333rem;
      cursor          : pointer;
      user-select     : none;
      background-color: var(--px-accordion-header-background-color, white);
      color           : var(--px-accordion-header-color, black);

      > * {
        user-select: none;
      }
    }

    &__icon {
      width : 1rem;
      height: 1rem;
    }

    &__body {
      margin  : 0;
      overflow: auto;
    }

    &__status {
      margin-right  : 1rem;
      text-transform: none;
    }
  }

  .accordion__expand__collapse {
    margin-bottom: 1.5rem;
  }

  &--disabled {
    .accordion__header {
      cursor        : not-allowed;
      pointer-events: none;
    }
  }
}