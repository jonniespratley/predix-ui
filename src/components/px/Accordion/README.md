The `<Accordion/>` component provides expandable and collapsible subsection headers for a page.


## Usage

```react
const { Accordion } = PxReact;
<div>
  <Accordion headerText="Header Caption" status="Last Updated: 3 Days Ago">
    Accordion content goes here.
  </Accordion>

  <Accordion headerText="Header Caption" status="Last Updated: 3 Days Ago">
    Accordion content goes here.
  </Accordion>
</div>
```

### With custom CSS

```react
const { Accordion } = PxReact;
<div className='custom'>
  <style>{`
    .custom {
      --px-accordion-header-color: white;
      --px-accordion-header-background-color: crimson;
    }
  `}</style>
  <Accordion
    headerValue="Header Caption"
    status="Last Updated: 3 Days Ago">
    <p>Accordion content goes here.</p>
  </Accordion>
</div>
```

## Properties

```table
span: 6
rows:
  - Name: disabled
    Type: bool
    Description: Flag indicating whether the accordion should be disabled.

  - Name: headerText | headerValue
    Type: string
    Description: String that will appear in header.
  
  - Name: status
    Type: string
    Description: String that will appear in the right hand side of the accordion.

  - Name: showAction
    Type: bool
    Description: Flag indicating whether the 'action' icon should be shown on the right hand side.
  
  - Name: opened
    Type: bool
    Description: If true the accordion is opened. If false it is closed. Use to observe the user opening/closing the accordion, or set to programatically change the accordion state.

  - Name: onCollapsed
    Type: func
    Description: Callback function when accordion is collapsed.
  
  - Name: onExpanded
    Type: func
    Description: Callback function when accordion is expanded.

  - Name: onActionClick
    Type: func
    Description: Callback function when action icon is clicked.
  
```

## Styling
The following custom properties are available for styling

```table
rows:
  - Custom Property: --px-headings-heading-subsection-color
    Description: Text color for the accordion header
  
  - Custom Property: --px-headings-heading-subsection-background
    Description: Background color for the accordion header
```
