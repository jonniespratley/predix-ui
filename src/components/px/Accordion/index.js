import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styled, { css } from 'styled-components';
import IronCollapse from '../../iron/IronCollapse';
import Flex from '../../../styles/flex';
import Icon from '../IconSet/Icon';

const styles = {
  width: '16px',
  height: '16px'
};
const svgStyles = {
  pointerEvents: 'none',
  display: 'block',
  height: '16px',
  width: '16px',
  fill: 'none',
  stroke: 'currentColor'
};
const CloseIcon = () => (<i className="px-icon px-icon-utl px-utl-chevron" style={styles} ><svg viewBox="0 0 16 16" preserveAspectRatio="xMidYMid meet" style={svgStyles}><g id="px-utl-chevron-right"><path strokeLinejoin="round" d="M6.2 13.2l5.4-5.5-5.5-5.5" /></g></svg></i>);
const OpenIcon = () => (<i className="px-icon px-icon-utl px-utl-chevron-down" style={styles}><svg viewBox="0 0 16 16" preserveAspectRatio="xMidYMid meet" style={svgStyles}><g id="px-utl-chevron"><path d="M2.4 6.2l5.5 5.5 5.5-5.5" /></g></svg></i>);

const AccordionHeader = styled.div`
  font-family: "GE Inspira Sans";
  padding: .5rem;
  margin-bottom   : 0.3333rem;
  cursor          : pointer;
  user-select     : none;

  background-color: var(--px-accordion-header-background-color, whitesmoke);
  color           : var(--px-accordion-header-color, black);

  display         : flex;
  justify-content : space-between;
  > * {
    user-select   : none;
  }

  ${props => props.disabled && css`
    cursor:not-allowed;
    pointerEvents: none;
  `}
`;
AccordionHeader.displayName = 'AccordionHeader';
AccordionHeader.defaultProps = {
  className: 'px-accordion__header'
};

const AccordionHeaderText = styled.span`
  text-transform:uppercase;
  margin-left: .2rem;
`;
AccordionHeaderText.displayName = 'AccordionHeaderText';

const AccordionContainer = styled.div`
  display: flex;
  flex-direction: column;
`;
AccordionContainer.displayName = 'AccordionContainer';
AccordionContainer.defaultProps = {
  className: 'px-accordion__container'
};

const AccordionSection = styled.div`
  display: flex;
  flex-direction: column;
`;
AccordionSection.displayName = 'AccordionSection';
AccordionSection.defaultProps = {
  className: 'px-accordion__section'
};

const AccordionBody = styled.div`
  margin  : 0;
  overflow: auto;
  padding: .5rem;
`;
AccordionBody.displayName = 'AccordionBody';
AccordionBody.defaultProps = {
  className: 'px-accordion__body'
};

const AccordionIcon = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width : 1rem;
  height: 1rem;
`;
AccordionIcon.displayName = 'AccordionIcon';

const AccordionStatus = styled.div`
  margin-right  : 1rem;
  text-transform: none;
`;
AccordionStatus.displayName = 'AccordionStatus';

const AccordionAction = styled.div`
  cursor: pointer;
  color: var(--px-actionable-alt-text-color, blue);
  background: transparent;
  border: transparent;
  outline: none;
  --iron-icon-width: 1rem;
  --iron-icon-height: 1rem;
`;
AccordionAction.displayName = 'AccordionAction';

class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: props.opened
    };
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    if (!this.props.disabled) {
      this.setState(prevState => ({
        open: !prevState.open
      }));
    }
  }

  componentDidUpdate() {
    const { onCollapsed, onExpanded } = this.props;
    if (this.state.open) {
      onExpanded ? onExpanded(this.state) : null;
    } else {
      onCollapsed ? onCollapsed(this.state) : null;
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ open: nextProps.opened });
  }

  render() {
    const { open } = this.state;
    const {
      headerText,
      headerValue,
      status,
      disabled,
      actions,
      showAction,
      onActionClick,
      icons,
      children
    } = this.props;
    const { action } = icons;

    const baseClasses = classNames(
      'px-accordion',
      { 'px-accordion--disabled': disabled },
      { 'px-accordion--open': open }
    );
    const headerClasses = classNames('px-accordion__header');

    return (
      <AccordionContainer className={baseClasses}>
        <AccordionHeader disabled={disabled} className={headerClasses}>
          <Flex item onClick={this.onClick} >
            <span>
              {open && <OpenIcon />}
              {!open && <CloseIcon />}
            </span>
            <AccordionHeaderText>{headerText || headerValue}</AccordionHeaderText>
          </Flex>
          <Flex middle>
            {status && <AccordionStatus>{status}</AccordionStatus>}
            {showAction &&
            <AccordionAction onClick={onActionClick}>
              <Icon icon={action} size={16} />
            </AccordionAction>}
            {actions &&
            <AccordionAction>
              {actions()}
            </AccordionAction>}
          </Flex>
        </AccordionHeader>
        <IronCollapse opened={open}>
          <AccordionBody>
            {children}
          </AccordionBody>
        </IronCollapse>
      </AccordionContainer>
    );
  }
}

Accordion.displayName = 'Accordion';

Accordion.propTypes = {
  headerText: PropTypes.string,
  headerValue: PropTypes.string,
  status: PropTypes.string,
  opened: PropTypes.bool,
  disabled: PropTypes.bool,
  actions: PropTypes.node,
  showAction: PropTypes.bool,
  onExpanded: PropTypes.func,
  onCollapsed: PropTypes.func,
  onActionClick: PropTypes.func,
  icons: PropTypes.shape({
    more: PropTypes.string,
    left: PropTypes.string,
    action: PropTypes.string
  }),
  children: PropTypes.node
};

Accordion.defaultProps = {
  headerText: null,
  headerValue: null,
  status: null,
  opened: true,
  disabled: null,
  actions: null,
  showAction: null,
  onCollapsed: null,
  onExpanded: null,
  onActionClick: null,
  icons: {
    more: 'px-utl:chevron-right',
    left: 'px-utl:chevron',
    action: 'px-utl:edit'
  },
  children: null
};

export default Accordion;
