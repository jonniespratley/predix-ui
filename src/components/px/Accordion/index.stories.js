import React from 'react';

import { storiesOf } from '@storybook/react';
import { withReadme } from 'storybook-readme';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs/react';

import Accordion from './';
import README from './README.md';

storiesOf('Accordion', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(README))
  .addWithJSX('default', () => (
    <Accordion
      onCollapsed={action('onCollapsed')}
      onExpanded={action('onExpanded')}
      onActionClick={action('onActionClick')}
      showAction={boolean('showAction', true)}
      opened={boolean('opened', true)}
      disabled={boolean('disabled', false)}
      status={text('status', 'Last Updated: 3 days ago')}
      headerValue={text('headerValue', 'Header Caption')}
    >
      {text('content', 'Accordion content goes here.')}
    </Accordion>
  ))
  .addWithJSX('with group', () => (
    <div>
      <Accordion
        showAction={boolean('showAction', true)}
        opened={boolean('opened', true)}
        disabled={boolean('disabled', false)}
        status={text('status', 'Last Updated: 3 days ago')}
        headerValue={text('headerValue', 'Header Caption')}
      >
        {text('content', 'Accordion content goes here.')}
      </Accordion>
      <Accordion
        showAction={boolean('showAction', true)}
        opened={boolean('opened', true)}
        disabled={boolean('disabled', false)}
        status={text('status', 'Last Updated: 3 days ago')}
        headerValue={text('headerValue', 'Header Caption')}
      >
        {text('content', 'Accordion content goes here.')}
      </Accordion>
    </div>
  ));

