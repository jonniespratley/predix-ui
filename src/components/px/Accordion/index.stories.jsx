import React from 'react';
import { storiesOf } from '@storybook/react';
import { withReadme } from 'storybook-readme';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean } from '@storybook/addon-knobs/react';

import withTests from '../../../../.storybook/withTests';

import Accordion from './';
import README from './README.md';

storiesOf('Components / Accordion', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(README))
  .addDecorator(withTests('Accordion'))
  .addWithJSX('default', () => (
    <Accordion
      onCollapsed={action('onCollapsed')}
      onExpanded={action('onExpanded')}
      onActionClick={action('onActionClick')}
      showAction={boolean('showAction', true)}
      opened={boolean('opened', true)}
      disabled={boolean('disabled', false)}
      status={text('status', 'Last Updated: 3 days ago')}
      headerValue={text('headerValue', 'Header Caption')}
    >
      {text('content', 'Accordion content goes here.')}
    </Accordion>
  ))
  .addWithJSX('with group', () => (
    <div>
      <Accordion
        showAction={boolean('showAction', true)}
        opened={boolean('opened', true)}
        disabled={boolean('disabled', false)}
        status={text('status', 'Last Updated: 3 days ago')}
        headerValue={text('headerValue', 'Header Caption')}
      >
        {text('content', 'Accordion content goes here.')}
      </Accordion>
      <Accordion
        showAction={boolean('showAction', true)}
        opened={boolean('opened', true)}
        disabled={boolean('disabled', false)}
        status={text('status', 'Last Updated: 3 days ago')}
        headerValue={text('headerValue', 'Header Caption')}
      >
        {text('content', 'Accordion content goes here.')}
      </Accordion>
    </div>
  ));

