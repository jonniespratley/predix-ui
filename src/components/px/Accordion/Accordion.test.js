import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import Accordion from './';

describe('Accordion', () => {
  test('should render open by default', () =>{
    const wrapper = shallow(
      <Accordion headerText="Header Caption" status="Last Updated: 3 Days Ago">
        <p>Accordion content goes here.</p>
      </Accordion>
    );
    expect(wrapper.find('.px-accordion__header').exists());
    expect(wrapper.state().open === true);
    wrapper.setProps({open: false})
    expect(wrapper.state().open === false);
    const tree = renderer.create(
      <Accordion headerText="Header Caption" status="Last Updated: 3 Days Ago">
        <p>Accordion content goes here.</p>
      </Accordion>
    ).toJSON();
      expect(tree).toMatchSnapshot();
  });

  test('should toggle open/closed', () =>{
    let isOpen = true;
    let onCollapsed = sinon.spy();
    let onExpanded = sinon.spy();
    const wrapper = shallow(
      <Accordion headerText='Should Toggle' open={isOpen} onCollapsed={onCollapsed} onExpanded={onExpanded}>
        <p>Accordion content goes here.</p>
      </Accordion>
    );
    expect(onExpanded.called);
    expect(wrapper.state().open === true);

    expect(wrapper.find('.px-utl-chevron-down').exists());
    wrapper.find('.px-accordion__header').simulate('click');
    expect(onCollapsed.called);
    expect(wrapper.state().open === false);
    expect(wrapper.find('.px-utl-chevron').exists());
  });

  test('can be disabled', () => {
    let spy = sinon.spy();
    const wrapper = shallow(
      <Accordion headerText='Should Toggle' onCollapsed={spy} disabled>
        <p>Accordion content goes here.</p>
      </Accordion>
    );
    expect(wrapper.find('.px-accordion--open').length).toEqual(1);
    wrapper.find('.px-accordion__header').simulate('click');
    expect(!spy.called);
    expect(wrapper.find('.px-accordion--open').length).toEqual(1);
    expect(wrapper.find('.px-accordion--disabled').length).toEqual(1);
  });
});
