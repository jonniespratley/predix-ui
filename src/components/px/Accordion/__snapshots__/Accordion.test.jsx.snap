// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render open by default 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  padding: .5rem;
  margin-bottom: 0.3333rem;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--px-headings-heading-subsection-color,black);
  background-color: var(--px-headings-heading-subsection-background,whitesmoke);
  background-color: var(--px-accordion-header-background-color,whitesmoke);
  color: var(--px-accordion-header-color,black);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 > * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  text-transform: uppercase;
  margin-left: .2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: "GE Inspira Sans";
  -webkit-transition-duration: 300ms;
  -moz-transition-duration: 300ms;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
}

.c6 {
  margin: 0;
  overflow: auto;
  padding: .5rem;
}

.c5 {
  margin-right: 1rem;
  text-transform: none;
}

<div
  className="px-accordion px-accordion--open c0"
>
  <div
    className="px-accordion__header c1"
    disabled={null}
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      <span>
        <i
          className="px-icon px-icon-utl px-utl-chevron-down"
          style={
            Object {
              "height": "16px",
              "width": "16px",
            }
          }
        >
          <svg
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "display": "block",
                "fill": "none",
                "height": "16px",
                "pointerEvents": "none",
                "stroke": "currentColor",
                "width": "16px",
              }
            }
            viewBox="0 0 16 16"
          >
            <g
              id="px-utl-chevron"
            >
              <path
                d="M2.4 6.2l5.5 5.5 5.5-5.5"
              />
            </g>
          </svg>
        </i>
      </span>
      <span
        className="c3"
      >
        Header Caption
      </span>
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        Last Updated: 3 Days Ago
      </div>
    </div>
  </div>
  <div
    aria-expanded={true}
    aria-hidden={false}
    className="iron-collapse"
  >
    <div
      className="px-accordion__body c6"
    >
      <p>
        Accordion content goes here.
      </p>
    </div>
    <style>
      
        .iron-collapse {
          display: block;
          -webkit-transition-duration: 300ms;
          -moz-transition-duration: 300ms;
          transition-duration: 300ms;
          overflow: visible;
        }
        .iron-collapse:not(.iron-collapse-opened) {
          overflow: hidden;
        }
      
    </style>
  </div>
</div>
`;
